lib/client/js/ui/controls/AuthButton.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * The UI Control that manages the AuthButton button component
+ * residing in the Header section
+ *
+ * @class
+ */
+var AuthButton = (function () {
+
+    /**
+     * @alias AuthButton
+     */
+    function AuthButton(config) {
+        // Call the super
+        Control.apply(this, [config]);
+    }
+
+    AuthButton.prototype = new Control();
+
+    AuthButton.prototype.view = function () {
+        if (this.getProp("loggedInUser") === null) {
```

Comment:
When you have a very simple decision point, I like to remove boilerplate via a ternary to make it simpler to read:

```
return this.getProp("loggedInUser") 
    ? $("<button>").text("Login") 
    : $("<button>").text("Logout");
```

Also, you don't need to explicitly check for `null`. It's usually enough just to check for truthiness.